src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@Nadahar to me the most readable and standard way is to use `if...else`:

```
if (getBoolean(KEY_OLD_PREVENTS_SLEEP, true)) {
	sleepMode = PreventSleepMode.PLAYBACK;
} else {
	sleepMode = PreventSleepMode.NEVER;
}
```

Since then the code is more like most human languages. So if we are picking a convention I would choose that.
I do like how ternaries save repetition and space though, they are more efficient.
The change you made in the new PR is an improvement but I agree with your comment about the question mark.

Personally I use `if...else` for multi-line, and ternary is single-line, but that's just me personally.